internal/slicer/slicer.go

```diff
@@ -153,12 +155,19 @@ func Run(options *RunOptions) error {
 		if reader == nil {
 			continue
 		}
-		err := deb.Extract(reader, &deb.ExtractOptions{
+		fileCreatorProxy := fsutil.NewFileCreatorProxy()
+		err := deb.Extract(fileCreatorProxy, reader, &deb.ExtractOptions{
 			Package:   slice.Package,
 			Extract:   extract[slice.Package],
 			TargetDir: targetDir,
 			Globbed:   globbedPaths,
 		})
+		for _, file := range fileCreatorProxy.Files {
+			err := report.AddFile(slice, file)
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
nit: since we are doing a similar routine in multiple places, it would be nice to move this to a function maybe. But I don't have any suggestions where it should be though.

Reply:
I see your point but I dont think creating a function is necessary. In this case we will be moving these lines and creating a function that takes `slice`, `fileCreatorProxy` and `report` and add the files internally. We would have to convey that with a good name and maybe a comment. I think these 5 lines do exactly that and it is easier to read the code this way.